### PR TITLE
PUB-1401 - MI Endpoints

### PR DESCRIPTION
**Pull request by furnivall**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-1401

### Change description ###
MI endpoints for subs man

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java [hunk 1]
```diff
@@ -31,6 +31,13 @@ public interface SubscriptionRepository extends JpaRepository<Subscription, Long
     List<Subscription> findSubscriptionsBySearchValue(@Param("search_type") String searchType,
                                                       @Param("search_value") String searchValue);
 
+    @Query(value = "SELECT cast(id as text), channel, search_type, user_id, court_name FROM subscription",
+        nativeQuery = true)
+    List<String> getAllSubsDataForMi();
+
+    @Query(value = "SELECT cast(ID as text), search_value, channel, user_id, court_name FROM SUBSCRIPTION "
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java:38]: The word "SUBSCRIPTION" in the query is in upper case whereas the other one is in lower case. Shall we make them consistent? :)

Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java]: Sorted

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java [hunk 2]
```diff
@@ -117,6 +117,26 @@ public ResponseEntity<String> buildDeletedArtefactSubscribers(@RequestBody Artef
...
+    @GetMapping("/mi-data-all")
+    public ResponseEntity<String> getSubscriptionDataForMiReportingAll() {
+        return ResponseEntity.status(HttpStatus.ACCEPTED)
+            .body(subscriptionService.getAllSubscriptionsDataForMiReporting());
+    }
+
+    @ApiResponses({
+        @ApiResponse(code = 202, message = "Subscription Management - MI Data request (local) accepted.")
+    })
+    @ApiOperation("Returns a list of subscription data for MI reporting.")
```
[the remark this answers is not in the document]
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java]: Sorted

**Review by KianKwa — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java [hunk 3]
```diff
@@ -117,6 +117,26 @@ public ResponseEntity<String> buildDeletedArtefactSubscribers(@RequestBody Artef
             "Deleted artefact third party subscriber notification request has been accepted");
     }
 
+    @ApiResponses({
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java:120]: I'm wondering if we need to take a date range in for these (or paginate the data in some way). A huge number os subs eventually could cause this endpoint to timeout if we have a huge number

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java (hunk 3 above)
Comment (furnivall) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java:120]: Yeah, that was the plan I had in mind - I was thinking throughout that parameterising could be added later so we could adapt to business reqs. Could either overload with params or require them - what do you reckon?

Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/controllers/SubscriptionController.java:120]: Overload makes sense I reckon!

Happy for this to be adapted / built upon later as the MI reporting grows

**Review by joshblackmoor — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java [hunk 4]
```diff
@@ -31,6 +31,13 @@ public interface SubscriptionRepository extends JpaRepository<Subscription, Long
     List<Subscription> findSubscriptionsBySearchValue(@Param("search_type") String searchType,
                                                       @Param("search_value") String searchValue);
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/repository/SubscriptionRepository.java:33]: Bug below from testing - court_name should be location_name. Currently producing exception

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**